text/0000-focus-behavior.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Focus Behavior
+status: DRAFTED
+created_at: 2020-01-15
+updated_at: 2020-01-15
+pr: https://github.com/salesforce/lwc-rfcs/pull/25
+---
+
+# Focus Behavior
+
+## Summary
+
+When discussing focus behavior, we need to consider the three ways you can
+focus on elements: click focus, programmatic focus, and sequential focus. In
+general, anything that is click focusable or sequentially focusable is
+programmatically focusable, and focus behavior for clicks and sequential
+navigation can depend on the user agent and system settings. For example,
+Safari will not apply focus on buttons when clicked, but it will apply focus on
+buttons when the focus method is invoked [1].
+
+As such, the set of click focusable and sequentially focusable elements is not
+something that we can know a priori. What we do know when observing our
+supported user agent matrix is that, the set of _potentially_ focusable
+elements is the same. This proposal makes the assumption that it is acceptable
+for the framework to introduce shadow semantics via various focus management
+polyfills as long as we don't prevent the user from reaching a focusable
+element.
+
+## Motivation
+
+Generally speaking, it is outside the scope of LWC to normalize different
+behaviors across its supported user agent matrix. However, in the case of focus
+delegation, the framework does need to make informed decisions to normalize
+focus behaviors in order to avoid a broken user experience. This is especially
+true in the case of sequential focus navigation.
+
+The primary goal of this proposal will be to minimize differences between its
+synthetic Shadow DOM and native Shadow DOM implementations to facilitate the
+transition of components to native Shadow DOM.
+
+## Detailed design
+
+For simplification, unless otherwise stated, it can be assumed that all
+components are delegating focus.
+
+For simplification, subsequent usage of the term "focusable" will refer to
+whether or not an element is programmatically focusable (i.e., whether or not
+an element can receive focus when its focus method is invoked or it has the
+`autofocus` attribute).
+
+```
+focusable = programmatically-focusable = click-focusable ∪ sequentially-focusable
+```
+
+### Focusable elements
+
+The set of sequentially focusable elements is currently defined by this
+[monster selector]. This has been sufficient for our needs so far but can

```

Comment:
Some minor issues with that selector:
* It's missing the focusable native elements `details` and `iframe`
* Also, the query currently looks for any element with `[href]`. This is meant to pick up `a` and `area` elements, but could accidentally pick up `base` and `link` elements, as well as any custom element that decided to expose an href attribute. Recommend tightening the query to `a[href]` and `area[href]`.

Reply:
Looks like the test page covers all your points except for `base`. I'll create an issue specifically for that element so that it doesn't slip through the cracks.